websites/D/Duolingo/presence.ts

```diff
@@ -1,296 +1,281 @@
+					break;
 
-			const phrases = document.querySelectorAll(
-				`body > ${"div > ".repeat(11)} .phrase`
-			);
-			if (phrases.length !== 0) {
-				return set(
-					`Reading ${Array.from(phrases)
-						.map(p => p.textContent)
-						.join(" ")}`
-				);
+				default:
+					presenceData.details = `Working on a ${language.name} lesson`;
 			}
-			return set("Reading a story");
-		}
-		case "mistakes-review":
-			return set("Reviewing past mistakes");
-		case "checkpoint": {
-			if (path.length < 4) return;
+			handleLesson();
+			break;
 
-			const checkpoint = parseInt(path[2], 10) + 1;
-			if (path[3] === "practice")
-				return set(`Practicing Checkpoint ${checkpoint}`);
-			else if (path[3] === "bigtest")
-				return set(`Taking the Checkpoint ${checkpoint} test`);
-			return;
-		}
-		// No default
-	}
+		//TODO: (not used as frequent. will add in next PR; should be soon)
+		// case "practice-hub":
+		// 	updateData();
+		// 	presenceData.details = `Choosing ${language.name} practice method`;
+		// 	handleLesson();
 
-	return false;
-}
+		// 	break;
 
-function checkStartingPages(path: string[]): boolean {
-	function set(state: string) {
-		presenceData.details = "Getting started";
-		presenceData.state = state;
-		return true;
-	}
-
-	switch (path[0]) {
-		case "register":
-			return set("Choosing a language");
-		case "welcome":
-			return set("Answering some questions");
 		case "placement":
-			return set("Taking the placement test");
-		// No default
-	}
+			setLang(path[2]);
+			presenceData.details = `Taking ${language.name} placement test`;
+			handleLesson();
+			break;
 
-	return false;
-}
+		case "leaderboard":
+			//TODO: slideshow
+			presenceData.details = "Viewing leaderboard";
+			break;
 
-function determineText(path: string[]) {
-	if (checkBasicPages(path)) return;
-	else if (checkLearningPages(path)) return;
-	else if (checkStartingPages(path)) return;
-}
+		case "quests":
+			//TODO: slideshow
+			presenceData.state = "Viewing quests";
+			break;
 
-presence.on("UpdateData", async () => {
-	if (!language.imageKey || !language.name) {
-		delete presenceData.smallImageKey;
-		delete presenceData.smallImageText;
-	} else {
-		presenceData.smallImageKey = language.imageKey;
-		presenceData.smallImageText = `Learning ${language.name}`;
+		case "shop":
+			presenceData.state = "Viewing shop";
+			break;
+
+		case "courses":
+			presenceData.details = "Viewing available courses";
+			presenceData.smallImageKey = "duo_globe";
+			presenceData.smallImageText = "viewing courses";
+			break;
+		case "enroll":
+			setLang(path[2]);
+			presenceData.details = `Viewing ${language.name} course`;
+			break;
+
+		case "profile":
+		case "u": {
+			const username = path[2],
+				displayName = document.querySelector(
+					'h1[data-test="profile-username"] span'
+				)?.textContent,
+				img =
+					(
+						document.querySelector(
+							`img[alt="${displayName}"]`
+						) as HTMLImageElement
+					)?.src ?? "duo",
```

Comment:
might be easier to use `querySelector<HTMLImageElement>(...)` then needing to `as` it

Reply:
Fixed in [c8f49b5](https://github.com/PreMiD/Presences/pull/7760/commits/c8f49b57b1b4110e494856fb59c12b0921b4819c)